src/lib.rs

```diff
@@ -207,6 +237,8 @@ impl MockableMqttClient for AsyncMqttClient {
 
 /// Configuration for the mqtt client.
 pub struct MqttConfig {
+    /// Schema of the mqtt broker (mqtt or mqtts)
+    pub mqtt_protocol: String,
```

Comment:
Turned this into an enum that is just specified like that in the examples. This lead to having to split up the unencrypted and encrypted examples though since its not a env parameter anymore. Not sure if thats the best...